src/aggregator/client/writer_mgr.go

```diff
@@ -188,23 +188,24 @@ func (mgr *writerManager) Write(
 	instance placement.Instance,
 	shardID uint32,
 	payload payloadUnion,
-) error {
+) (int, error) {
 	mgr.RLock()
+
 	if mgr.closed {
-		mgr.RUnlock()
-		return errInstanceWriterManagerClosed
+		mgr.RUnlock() // Manually unlock read lock
+		return 0, errInstanceWriterManagerClosed
 	}
+
 	id := instance.ID()
 	writer, exists := mgr.writers[id]
+	mgr.RUnlock() // Manually unlock read lock
+
 	if !exists {
-		mgr.RUnlock()
-		return fmt.Errorf("writer for instance %s is not found", id)
+		return 0, fmt.Errorf("writer for instance %s not found", id)
 	}
-	writer.dirty.Store(true)
-	err := writer.Write(shardID, payload)
-	mgr.RUnlock()
 
-	return err
+	writer.dirty.Store(true)
```

Comment:
i don't know if the code relies on this semantic or not - but earlier we held the read lock when marking the dirty bit - we no longer do that after your change.

can you ensure that change is safe / if you're not sure - go back to the old way.

Reply:
it was outside of the scope or read lock, but I reverted that optimization to avoid any unknowns